include/ensmallen_bits/moead/moead_impl.hpp

```diff
@@ -0,0 +1,453 @@
+/**
+ * @file moead_impl.hpp
+ * @authors Utkarsh Rai, Nanubala Gnana Sai
+ *
+ * Implementation of the MOEA/D-DE algorithm. Used for multi-objective
+ * optimization problems on arbitrary functions.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more Information.
+ */
+
+#ifndef ENSMALLEN_MOEAD_MOEAD_IMPL_HPP
+#define ENSMALLEN_MOEAD_MOEAD_IMPL_HPP
+
+#include "moead.hpp"
+#include <assert.h>
+
+namespace ens {
+
+inline MOEAD::MOEAD(const size_t populationSize,
+                    const size_t maxGenerations,
+                    const double crossoverProb,
+                    const size_t neighborSize,
+                    const double distributionIndex,
+                    const double neighborProb,
+                    const double differentialWeight,
+                    const size_t maxReplace,
+                    const arma::vec& lowerBound,
+                    const arma::vec& upperBound) :
+    populationSize(populationSize),
+    maxGenerations(maxGenerations),
+    crossoverProb(crossoverProb),
+    neighborSize(neighborSize),
+    distributionIndex(distributionIndex),
+    neighborProb(neighborProb),
+    differentialWeight(differentialWeight),
+    maxReplace(maxReplace),
+    lowerBound(lowerBound),
+    upperBound(upperBound),
+    numObjectives(0)
+  { /* Nothing to do here. */ }
+
+inline MOEAD::MOEAD(const size_t populationSize,
+                    const size_t maxGenerations,
+                    const double crossoverProb,
+                    const size_t neighborSize,
+                    const double distributionIndex,
+                    const double neighborProb,
+                    const double differentialWeight,
+                    const size_t maxReplace,
+                    const double lowerBound,
+                    const double upperBound) :
+    populationSize(populationSize),
+    maxGenerations(maxGenerations),
+    crossoverProb(crossoverProb),
+    neighborSize(neighborSize),
+    distributionIndex(distributionIndex),
+    neighborProb(neighborProb),
+    differentialWeight(differentialWeight),
+    maxReplace(maxReplace),
+    lowerBound(lowerBound * arma::ones(1, 1)),
+    upperBound(upperBound * arma::ones(1, 1)),
+    numObjectives(0)
+  { /* Nothing to do here. */ }
+
+//! Optimize the function.
+template<typename MatType,
+         typename... ArbitraryFunctionType,
+         typename... CallbackTypes>
+typename MatType::elem_type MOEAD::Optimize(std::tuple<ArbitraryFunctionType...>& objectives,
+                                            MatType& iterate,
+                                            CallbackTypes&&... callbacks)
+{
+  // Sanity checks
```

Comment:
```suggestion
  // Sanity checks.
```

Missing stop at the end.